File: extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin-serialization-common/runtime/src/main/java/io/quarkus/resteasy/reactive/kotlin/serialization/common/runtime/KotlinSerializationConfig.java
package io.quarkus.resteasy.reactive.kotlin.serialization.common.runtime;

import java.util.StringJoiner;

import io.quarkus.runtime.annotations.ConfigItem;
import io.quarkus.runtime.annotations.ConfigPhase;
import io.quarkus.runtime.annotations.ConfigRoot;

@ConfigRoot(name = KotlinSerializationConfig.CONFIG_NAME, phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
public class KotlinSerializationConfig {
    public static final String CONFIG_NAME = "kotlin-serialization";

    /**
     * Configuration element for serializing to json
     */
    @ConfigItem(name = "json")
    public JsonConfig json = new JsonConfig();

    @Override
    public String toString() {
        return new StringJoiner(", ", KotlinSerializationConfig.class.getSimpleName() + "[", "]")
                .add("json=" + json)
                .toString();
    }
}

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin-serialization/runtime/src/main/kotlin/io/quarkus/kotlin/serialization/KotlinSerializationConfig.kt
package io.quarkus.kotlin.serialization

import io.quarkus.runtime.annotations.ConfigRoot
import io.quarkus.runtime.annotations.ConfigItem
import io.quarkus.runtime.annotations.ConfigPhase.BUILD_AND_RUN_TIME_FIXED

@ConfigRoot(name = KotlinSerializationConfig.CONFIG_NAME, phase = BUILD_AND_RUN_TIME_FIXED)
class KotlinSerializationConfig {
    companion object {
        const val CONFIG_NAME = "kotlin-serialization"
    }

    /**
     * Configuration element for serializing to json
     */
    @ConfigItem(name = "json")
    @JvmField
    var json: JsonConfiguration = JsonConfiguration()

    override fun toString(): String {
        return "KotlinSerializationConfig(json=$json)"
    }
}